File: USACO/2014/JAN BRONZE/skicourse.py
import sys

sys.stdin = open("skidesign.in", "r")
sys.stdout = open("skidesign.out", "w")

t = sys.stdin.readline

def cost_calculate(n, k, maximum, minimum):
    cost = 0
    for i in range(n):
        if k[i] > maximum:
            cost += ((maximum - k[i])**2)
        if k[i] < minimum:
            cost += ((k[i]-minimum) **2)
    return cost

def solution(n, k):
    answer = float('inf')
    for i in range(84):
        answer = min(answer, cost_calculate(n, k, i+17, i))
    return answer


n = int(t())
temp_list = []
for i in range(n):
    k = int(t())
    temp_list.append(k)

print(solution(n, temp_list))

File: USACO/2014/JAN BRONZE/skicourse.cpp
#include <bits/stdc++.h>
using namespace std;

int elevation[1001];
int ans = 1e9;

int main() {

    freopen("skidesign.in", "r", stdin);
	freopen("skidesign.out", "w", stdout);

	int n;
	cin >> n;

	for (int i = 0; i < n; i++) {
		cin >> elevation[i];
	}
	
	for(int i = 0; i < 83; i++){

		int temp = 0;

		for (int j = 0; j < n; j++) {

			if(elevation[j] < i){
                temp += (i - elevation[j]) * (i - elevation[j]);
            }

			if(elevation[j] > i + 17){
                temp += (elevation[j] - i - 17) * (elevation[j] - i - 17);
            }
		}
		ans = min(ans, temp);
	}
	cout << ans;
}